src/main/scala/com/kinja/amqp/AmqpConfiguration.scala

```diff
@@ -26,6 +24,14 @@ final case class ResendLoopConfig(
 	memoryFlushTimeOut: FiniteDuration
 )
 
+object DeliveryGuarantee {
+	class Guarantee
+	case object AtLeastOnce extends Guarantee
+	case object AtMostOnce extends Guarantee
+}
```

Comment:
Is there a reason this is in an object? Also, you could use a sealed trait instead of a class.